src/parser/parser.yy

```diff
@@ -996,9 +999,6 @@ vid
         $$ = new ConstantExpression(*$1);
         delete $1;
     }
-    | legal_integer {
```

Comment:
Why change this?

Reply:
`legal_integer` is a positive integer, `int vid` can be a negative integer.

Reply:
Ok.